### PR TITLE
Password validation

### PR DESCRIPTION
**Pull request by kili236**
[Recap of our workflow](https://github.com/softozor/shopozor-backend#pull-requests)

# Prerequisites

- [ ] You have configured the triggering of the pre-commit hook on your local repository's clone. You need to have installed the python module `pre-commit` (listed in [the required dev modules](requirements-dev.txt)) and run the command `pre-commit install` in the repository's root folder.
- [ ] You have covered your code with unit and acceptance tests

# Changes

The password validation feature has been implemented according to the following rules:
    - il contient au moins 8 caractères
    - il mélange des chiffres et des lettres
    - il contient au moins un caractère spécial
    - il n'a pas été compromis

# How to use the feature

To activate the feature, you need to use the following command, with the wanted validator in the `settings.py` file

```
AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': '<myClassValidator>',
    },
]
```

# Additional notes

The program will probably block if no response from `https://haveibeenpwned.com/` is received.

### CODE REVIEW

**Comment by zadigus** [comment]
test this please

**Comment by softozor-deployer** [comment]
Can one of the admins verify this patch?

**Comment by zadigus** [comment]
ok to test

**Comment by zadigus** [comment]
ok to test

**Comment by zadigus** [comment]
add to whitelist

**Comment by zadigus** [comment]
test this please

**Comment by zadigus** [comment]
bon on a un petit problème avec le serveur de builds apparemment, je vais voir tout ça

**Comment by zadigus** [comment]
@kili236 please get rid of tag `@wip` from [feature file](https://github.com/shopozor/backend/blob/password-validation/features/Authentication/PasswordPolicy.feature).

**Comment by zadigus** [comment]
test this please

**Comment by zadigus** [comment]
ok to test

**Comment by zadigus** [comment]
ok to test

**Comment by zadigus** [comment]
ok to test

**Comment by zadigus** [comment]
on avance on avance; en fait suite à la migration des repos sur la nouvelle organisation `shopozor`, j'ai oublié d'ajouter l'utilisateur `softozor-deployer`, donc le serveur jenkins ne pouvait plus clone le repo

maintenant j'ai un autre problème que je vérifierai demain

**Comment by zadigus** [comment]
so, the test results are fine; the failure is the usually failing login scenario

**Comment by zadigus** [comment]
ok to test

File: unit_testing.groovy [hunk 1]
```diff
@@ -15,7 +15,10 @@ pipeline {
       steps {
         withEnv(["HOME=$WORKSPACE"]) {
           sh "virtualenv $VENV"
+          sh ". $VENV/bin/activate && pip install dos2unix"
```
Comment (donnerc) [inline unit_testing.groovy]: why install dos2unix ??? git has an integrated feature to do that automatically on commit on windows machines.

File: shopozor/password_validation.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        password = "".join(password.split())
+        if not re.findall('[^A-Za-z0-9]', password):
+            raise ValidationError(
+                _("This password must contain at least a special character."),
+                code='special_character_missing',
+            )
+
+    def get_help_text(self):
+        return _(
+            "Your password must contain at least a special character."
```
[the remark this answers is not in the document]
Comment (zadigus) [reply shopozor/password_validation.py]: @kili236 dry = Don't Repeat Yourself

File: unit_testing.groovy (hunk 1 above)
Comment (zadigus) [reply unit_testing.groovy]: not configured yet on that branch (but configured in the `dev` branch); can't tell yet if the `dev` branch config is ok

File: shopozor/password_validation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zadigus) [reply shopozor/password_validation.py]: you should put that error string in a member variable

**Comment by zadigus** [comment]
ok to test

**Comment by zadigus** [comment]
ok to test

**Comment by zadigus** [comment]
tests are ok

**Pull request merged**